compiler/backend_inkwell/src/lib.rs

```diff
@@ -0,0 +1,1041 @@
+
+                    let function_ctx_name =
+                        function_ctx.function_value.get_name().to_str().unwrap();
+                    if idx == mir.expressions.len() - 1 {
+                        if function_ctx_name != "main" {
+                            self.builder
+                                .build_return(Some(&struct_value.into_pointer_value()));
+                        } else {
+                            self.main_return.replace(struct_value.into_pointer_value());
+                        }
+                    }
+                }
+                candy_frontend::mir::Expression::Reference(ref_id) => {
+                    let value = self.get_value_with_id(function_ctx, ref_id).unwrap();
+
+                    self.locals.insert(*id, Rc::new(value));
+                    if idx == mir.expressions.len() - 1 {
+                        self.builder.build_return(Some(&value));
+                    }
+                }
+                candy_frontend::mir::Expression::HirId(hir_id) => {
+                    let i32_type = self.context.i32_type();
+                    let i8_type = self.context.i8_type();
+                    let candy_type_ptr = self
+                        .module
+                        .get_struct_type("candy_type")
+                        .unwrap()
+                        .ptr_type(AddressSpace::default());
+
+                    let text = format!("{hir_id}");
+
+                    let global = self.module.add_global(candy_type_ptr, None, &text);
+                    let v = self.make_str_literal(&text);
+                    let len = i32_type.const_int(text.len() as u64, false);
+                    let arr_alloc = self.builder.build_array_alloca(i8_type, len, "");
+                    self.builder.build_store(arr_alloc, v);
+                    let cast = self.builder.build_bitcast(
+                        arr_alloc,
+                        i8_type.ptr_type(AddressSpace::default()),
+                        "",
+                    );
+                    let make_candy_text = self.module.get_function("make_candy_text").unwrap();
+                    let call = self.builder.build_call(make_candy_text, &[cast.into()], "");
+
+                    self.builder.build_store(
+                        global.as_pointer_value(),
+                        call.try_as_basic_value().unwrap_left(),
+                    );
+
+                    global.set_initializer(&candy_type_ptr.const_null());
+                    self.values.insert(*id, Rc::new(global));
+                }
+                candy_frontend::mir::Expression::Function {
+                    original_hirs,
+                    parameters,
+                    body,
+                    responsible_parameter,
+                } => {
+                    let name = format!("{original_hirs:?}")
+                        .replace('{', "")
+                        .replace('}', "")
+                        .replace([':', '.'], "_");
```

Comment:
More explicit and with sorting:

```suggestion
                    let name = original_hirs
                        .iter()
                        .sorted()
                        .map(|it| it.to_string().replace([':', '.'], "_"))
                        .join(", ");
```

(Requires `itertools`)